### PR TITLE
Join a event (user request + withdraw; creator approve/reject + remove)

### PR DESCRIPTION
**Pull request by WenjunYuAnny**
Creator of the event:
creator approve a request
(cannot approve more people than capacity)
creator_approve limit

Logged in user:
user after log in
user after request
User not logged in:
not log in

### CODE REVIEW

**Review by Dhiren-NYU — CHANGES_REQUESTED** [review]
90,000 static file changes? No one will be able to review just because of the amount of files
Lets remove the static files
and change travis.yml so that collectstatic is run before deploying

File: events/templates/events/event-detail.html [hunk 1]
```diff
@@ -8,45 +8,150 @@
...
+            <div class="event-detail-container">
+                <div class="sticky-note-container">
+                    <div class="my-3"><h1>{{ event.event_name }}</h1></div>
+                    <h4>Creator</h4>
+                    <p>{{ event.creator }}</p>
+                    <h4>Date and Time</h4>
+                    <p>from {{ event.start_time }} to {{ event.end_time }}</p>
+                    <!-- <img src="event-image.jpg" alt="Event Image" class="img-fluid rounded"> -->
+                    <h4>Location</h4>
+                    <p>{{ event.event_location }}</p>
```
[the remark this answers is not in the document]
Comment (WenjunYuAnny) [reply events/templates/events/event-detail.html]: Yes I've adjusted this. Right now the event detail page also shows the address and the category of the location

File: events/templates/events/event-detail.html [hunk 2]
```diff
@@ -8,45 +8,150 @@
...
+                    <h4>Capacity</h4>
+                    <p>{{ event.capacity }}</p>
+                    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam eget tortor in eros euismod dignissim. Vivamus posuere,
```
[the remark this answers is not in the document]
Comment (WenjunYuAnny) [reply events/templates/events/event-detail.html:79]: Exactly!

File: events/templates/events/event-detail.html [hunk 3]
```diff
@@ -8,45 +8,150 @@
...
+            {% if user.is_authenticated %}
+                {% if user != event.creator %}
+                    <!-- The user is not the creator of the event -->
+                    {% if join_status == "pending" %}
+                        <!-- Button to withdraw request -->
+                        <form action="{% url 'events:toggle-join-request' event.id %}" method="post">
+                            {% csrf_token %}
+                            <button type="submit" class="btn btn-outline-primary btn-block join-button" id="withdraw-to-join-button" onclick="return confirm('Are you sure you want to withdraw your request?');">Withdraw the Request</button>
+                        </form>   
+                    {% elif join_status == "withdrawn" or join_status == None or join_status == "rejected" or join_status == "removed" %}
```
[the remark this answers is not in the document]
Comment (WenjunYuAnny) [reply events/templates/events/event-detail.html]: Yes I've done this in the code now

File: events/templates/events/event-detail.html [hunk 4]
```diff
@@ -8,45 +8,150 @@
...
+                        <!-- Button to join again or create a new EventJoin object-->
+                        <form action="{% url 'events:toggle-join-request' event.id %}" method="post">
+                            {% csrf_token %}
+                            <button type="submit" class="btn btn-primary btn-block join-button" id="request-to-join-button" onclick="return confirm('Ready to send a request to join?');">Request to Join</button>
```
[the remark this answers is not in the document]
Comment (WenjunYuAnny) [reply events/templates/events/event-detail.html]: Sure I've removed it

**Review by Dhiren-NYU — APPROVED** [review]

**Pull request merged**